### DISCUSSION TITLE
How to preserve HDR bit depth when doing convert

### DISCUSSION THREAD

**Discussion by tuket** [ImageMagick/ImageMagick#3401] [Help]
Hi,
I have a bunch of `.hdr` textures that have 16-bit float pixels.
I would like to collect all of those files into a single `.dds`. This is how I'm doing it:

```
$ magick convert "radiance.mip*.hdr" -define dds:mipmaps=fromlist "radians.dds"
```

It works, but the problem is that the output `.dds` file doesn't seem to preserve the 16-bit float format. What can I do?
Thanks!

**Comment by fmw42** [comment]
What is your IM version? You need a version that is HDRI compiled?  IM 7 is by default. But you should not add convert to the command. It will cause IM 6 behavior.  So use magick, not magick convert. Check `magick -version` to be sure it is HDRI. But I am not sure DDS format will support 16-bit floats.

**Reply by tuket** [reply]
Hi, thanks for your answer.

```
magick.exe --version
Version: ImageMagick 7.0.11-3 Q16 x64 2021-03-06 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Visual C++: 192829337
Features: Cipher DPC Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raw rsvg tiff webp xml zip zlib
```

The DDS format does support float textures.

I have found that, even when doing a `-resize` with `magick`, it will clamp values to a maximum of `1.0`. So there must be something I'm missing.

```
magick radiance.hdr -resize 1024x512 radiance.hdr
```

**Reply by fmw42** [reply]
Your version of ImageMagick is not HDRI. You will need to reinstall with HDRI enabled to be able to process float images. For example my version is

```
magick -version

Version: ImageMagick 7.0.11-3 Q16 x86_64 2021-03-06 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(3.1) 
Delegates (built-in): bzlib cairo djvu fftw fontconfig freetype gslib gvc jbig jng jp2 jpeg lcms lqr ltdl lzma openexr png ps raw rsvg tiff webp x xml zlib

```

Note the HDRI listed on the line of Features.

**Reply by tuket** [reply]
Thank you!
I seems to work for `-resize`.
However, I'm still having the same problem when generating the `.dds` file.

**Reply by fmw42** [reply]
Your ImageMagick does not seem to be compiled with HDRI enabled?  Where did you get your install and how did you install it (binary or from source)?

**Reply by tuket** [reply]
Yeah, I forgot to mention in my previous comment that I have downloaded the correct one now. With the right executable, now it preserves the HDRI information when doing `-resize`.

However, when converting to .dds, values higher than one are clamped. But I think the problem might be (as you initially suggested) that .dds doesn't support floats :(
At least I can't find it in the docs: https://docs.microsoft.com/en-us/windows/win32/direct3ddds/dds-pixelformat

**Reply by tuket** [reply]
I have found that it is supported. I have found one project that can generate them: https://github.com/dariomanesku/cmft/blob/cad5f31bac66fd05987d667af62311c444df6d46/src/cmft/image.cpp#L879

**Comment by fmw42** [comment]
But i do not know if IM supports it.
